src/_utils.js

```diff
@@ -650,6 +655,7 @@ export const setStateOptions = state => {
       vendorPrefixes: state.opts.vendorPrefixes
     })
   }
+  state.styleComponent = STYLE_COMPONENT
```

Comment:
```suggestion
```

The identifier should be generated in the `Program` visitor as a unique id https://github.com/vercel/styled-jsx/blob/master/src/babel.js#L286 and https://github.com/vercel/styled-jsx/blob/master/src/babel-external.js#L240

```js
state.styleComponentImportName = path.scope.generateUidIdentifier(STYLE_COMPONENT).node.name
```

Reply:
What about macro? It seems that macros cannot get the state property set in `Program` visitor.

Reply:
Is this still an issue?

Reply:
not anymore, will continue the import injection work later. thanks!